src/libsync/creds/oauth.cpp

```diff
@@ -557,4 +549,70 @@ void OAuth::openBrowser()
     });
 }
 
+void OAuth::dynamicRegistrationDataReceived(const QVariantMap &dynamicRegistrationData)
+{
+    // the default class (used by the wizard only) doesn't know how to handle such data, and also doesn't need to
+    // only the derived AccountBasedOAuth class must be able to handle the data
+    (void)dynamicRegistrationData;
+}
+
+
+AccountBasedOAuth::AccountBasedOAuth(AccountPtr account, QObject *parent)
+    : OAuth(account->url(), account->davUser(), account->networkAccessManager(), {}, parent)
+    , _account(account)
+{
+}
+
+void AccountBasedOAuth::startAuthentication()
+{
+    auto credentialsJob = _account->credentialManager()->get(dynamicRegistrationDataC());
+
+    connect(credentialsJob, &CredentialJob::finished, this, [=] {
+        credentialsJob->deleteLater();
+
+        qDebug() << credentialsJob->errorString() << credentialsJob->error();
```

Comment:
Debug leftover

Reply:
I didn't mind having this debug snippet in our code base. But I'll remove it.